File: c/puzzles/toString.c
#include <stdio.h>

void reverse(char *s) {
    int c, i, j;

    for (j = 0; s[j] != '\0'; j++)
        ;

    for (i = 0, j--; i < j; i++, j--) {
        c = s[i];
        s[i] = s[j];
        s[j] = c;
    }
}

int *toString(int n, char *buf) {
    int k = 0;

    do {
        buf[k++] = (n % 10) + '0';
    } while (n /= 10);

    buf[k] = '\0';
    reverse(buf);
}

int main(int argc, char **argv) {
    char buf[255];
    toString(732, buf);
    printf("%s\n", buf);

    return 0;
}

File: javascript/puzzles/toString.js
const reverse = s => {
    const a = s.split('');
    let i, j, c;

    for (i = 0, j = a.length - 1; i < j; i++, j--) {
        c = a[i];
        a[i] = a[j];
        a[j] = c;
    }

    return a.join('');
};

const toString = n => {
    let i = '';

    do {
        i += String.fromCodePoint((n % 10) + '0'.codePointAt(0));
    } while ((n = n / 10 >> 0) > 0);

    return reverse(i);
};

console.log(
    toString(7115)
);

